components/scream/src/physics/shoc/atmosphere_macrophysics.cpp

```diff
@@ -70,61 +73,49 @@ void SHOCMacrophysics::set_grids(const std::shared_ptr<const GridsManager> grids
+  m_required_fields.emplace("wthv_sec",   scalar3d_layout_mid, K*(m/s),     grid_name);
+  m_required_fields.emplace("tkh",        scalar3d_layout_mid, (m*m)/s,     grid_name);
+  m_required_fields.emplace("tk",         scalar3d_layout_mid, (m*m)/s,     grid_name);
+  m_required_fields.emplace("shoc_ql",    scalar3d_layout_mid, Qunit,           grid_name);
+
+  m_computed_fields.emplace("s",          scalar3d_layout_mid, J/kg,        grid_name);
+  m_computed_fields.emplace("tke",        scalar3d_layout_mid, (m*m)/(s*s), grid_name);
+  m_computed_fields.emplace("horiz_wind", horiz_wind_layout,   m/s,         grid_name);
+  m_computed_fields.emplace("wthv_sec",   scalar3d_layout_mid, K*(m/s),     grid_name);
+  m_computed_fields.emplace("tkh",        scalar3d_layout_mid, (m*m)/s,     grid_name);
+  m_computed_fields.emplace("tk",         scalar3d_layout_mid, (m*m)/s,     grid_name);
+  m_computed_fields.emplace("shoc_ql",    scalar3d_layout_mid, Qunit,       grid_name);
 
   // Output variables
   m_computed_fields.emplace("pblh", scalar2d_layout_col, m, grid_name);
 
   // Tracer group
   m_inout_groups_req.emplace("TRACERS",grid->name());
-
-  // Wind group
-  m_inout_groups_req.emplace("horiz_wind",grid->name());
 }
 // =========================================================================================
 void SHOCMacrophysics::
 set_updated_group (const FieldGroup<Real>& group)
 {
+  EKAT_REQUIRE_MSG(group.m_info->size() >= 3,
+                   "Error! Shoc requires at least 3 tracers (tke, shoc_qv, shoc_ql) as inputs.");
+
   const auto& name = group.m_info->m_group_name;
 
-  EKAT_REQUIRE_MSG(name=="TRACERS" || name=="horiz_wind",
+  EKAT_REQUIRE_MSG(name=="TRACERS",
     "Error! We were not expecting a field group called '" << name << "\n");
 
   EKAT_REQUIRE_MSG(group.m_info->m_bundled,
-      "Error! Shoc expects bundled fields for tracers and wind.\n");
-
-  if (name=="TRACERS") {
-    EKAT_REQUIRE_MSG(group.m_info->size() >= 3,
-                     "Error! Shoc requires at least 3 tracers (tke, shoc_qv, shoc_ql) as inputs.");
-
-    // Add Q bundle as in/out field
-    m_shoc_fields_in["Q"]  = *group.m_bundle;
-    m_shoc_fields_out["Q"] = *group.m_bundle;
+      "Error! Shoc expects bundled fields for tracers.\n");
 
-    // Calculate number of advected tracers
-    m_num_tracers = m_shoc_fields_in["Q"].get_header().get_identifier().get_layout().dim(1);
-  }
-
-  if (name=="horiz_wind") {
-    EKAT_REQUIRE_MSG(group.m_info->size() == 2,
-                     "Error! Shoc requires exactly 2 wind vectors (u, v) as inputs.");
+  // Add Q bundle as in/out field
+  m_shoc_fields_in["Q"]  = *group.m_bundle;
+  m_shoc_fields_out["Q"] = *group.m_bundle;
 
-    // Add V bundle as in/out field
-    m_shoc_fields_in["V"]  = *group.m_bundle;
-    m_shoc_fields_out["V"] = *group.m_bundle;
-  }
+  // Calculate number of advected tracers
+  m_num_tracers = m_shoc_fields_in["Q"].get_header().get_identifier().get_layout().dim(1);
```

Comment:
This is perfectly fine, but for the record, another way (perhaps more verbose) is
```
m_num_tracers = group.m_info->size();
```

Reply:
Yes, much better.